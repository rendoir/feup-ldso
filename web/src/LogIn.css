div#login-container {
    /* transform-origin: center center;
    position: absolute;
    margin-left: auto;
    margin-right: auto;
    margin: auto auto;
    left: 0;
    right: 0;
    top: 0;
    bottom: 0; */
    background-color: white;
    height: 100vh;
}

div.login-col-first {
    background-color: #292929;
}

div.login-col-first img {
    transform-origin: center center;
    position: absolute;
    margin-left: auto;
    margin-right: auto;
    left: 0;
    right: 0;
    top: 25%;
    max-width: 50%;
}

div#login-container div.row {
    /* transform-origin: center center;
    position: absolute;
    margin-left: auto;
    margin-right: auto;
    left: 0;
    right: 0;
    top: 25%; */
    height: 100%;
}

div#login-container form {
    transform-origin: center center;
    position: absolute;
    margin-left: auto;
    margin-right: auto;
    left: 0;
    right: 0;
    top: 40%;
    max-width: 50%;
    background-color: white;
}

span#basic-addon {
    padding: 0.5em;
    max-height: 33.5px;
    border: 1px solid #455a64;
    background-color: white;
}

div#login-container form h2 {
    margin-bottom: 1em;
}

div#login-container form input {
    margin-bottom: 1em;
    border: 1px solid #455a64;
    border-left: 0;
}

button#login-button {
    background-color: #455a64;
    border-color: #455a64;
    border-radius: 50px;
    font-size: 16px;
    font-weight: 600;
    padding: 0.5em 1.5em;
    color: white;
}