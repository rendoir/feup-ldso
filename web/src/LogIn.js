import React, { Component } from 'react';
import { Redirect } from 'react-router-dom';
import { Form, FormControl, Col, Button, Row, Alert } from 'react-bootstrap';
import axios from 'axios';

class LogIn extends Component {

    constructor(props) {
        super(props);

        this.state = {
            emailInput: "",
            passwordInput: "",
            alertType: null,
            alertMessage: null
        };

        this.updateEmailInput = this.updateEmailInput.bind(this);
        this.updatePasswordInput = this.updatePasswordInput.bind(this);
        this.handleLogIn = this.handleLogIn.bind(this);
    }

    updateEmailInput(event) {
        this.setState({ emailInput: event.target.value });
    }

    updatePasswordInput(event) {
        this.setState({ passwordInput: event.target.value });
    }

    handleLogIn(event) {
        event.preventDefault();

        let self = this;

        axios.post('http://localhost:3030/login', {
            email: self.state.emailInput,
            password: self.state.passwordInput
        })
            .then((res) => {
                document.cookie = 'access_token=' + res.data.token;
                self.setState({ alertType: 'success', alertMessage: "Successfully logged in!" });
            })
            .catch((err) => {
                self.setState({ alertType: 'danger', alertMessage: "Error logging in: " + err.response.data });
            });
    }

    render() {

        let alertElement;
        if (this.state.alertMessage !== null) {
            alertElement = (
                <Row>
                    <Col sm={4} md={2}>

                    </Col>
                    <Col sm={5} md={8}>
                        <Alert className={this.state.alertType}>
                            {this.state.alertMessage}
                        </Alert>
                    </Col>
                    <Col sm={4} md={2}>

                    </Col>
                </Row>);
        }

        if (document.cookie === undefined ||
            document.cookie.indexOf("access_token=") !== -1) return <Redirect to={'/events'} />;

        return (
            <div id="login-container">
                {alertElement}
                <Row>
                    <Col sm={4}></Col>
                    <Col sm={4}>
                        <Form onSubmit={this.handleLogIn}>
                            <FormControl type="email" required placeholder="Email"
                                value={this.state.emailInput} onChange={this.updateEmailInput} />

                            <FormControl type="password" required placeholder="Password"
                                value={this.state.passwordInput} onChange={this.updatePasswordInput} />

                            <Button variant="primary" type="submit"
                                id="login-button" className="primary_button">Entrar</Button>
                        </Form>
                    </Col>
                    <Col sm={4}></Col>
                </Row>
            </div>
        );

    }

}

export default LogIn;
