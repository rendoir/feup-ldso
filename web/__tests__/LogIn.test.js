import React from 'react';
import rendered from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import LogIn from '../src/LogIn';
import { mount } from 'enzyme';
import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';

var mockAxios = new MockAdapter(axios);

document.cookie = undefined;

describe("Check Render ListEvents", () => {

    it('renders log in form', () => {

        const listEvents = rendered.create(
            <BrowserRouter>
                <LogIn />
            </BrowserRouter>
        );
        let tree = listEvents.toJSON();
        expect(tree).toMatchSnapshot();
    });

});

describe("Check Update Functions", () => {

    var wrapper;

    beforeAll(async() => {
        document.cookie = undefined;

        wrapper = await mount(
            <BrowserRouter>
                <LogIn />
            </BrowserRouter>);
    });


    it('check update email input', () => {

        let wrapperLogIn = wrapper.find(LogIn).first();

        const input = wrapperLogIn.find("input.email-input").first();
        input.simulate('change', {
            target: { value: 'email@email.com' }
        });

        wrapperLogIn.update();
        expect(wrapperLogIn.state().emailInput).toEqual('email@email.com');

    });

    it('check update password input', () => {

        let wrapperLogIn = wrapper.find(LogIn).first();

        const input = wrapperLogIn.find("input.password-input").first();
        input.simulate('change', {
            target: { value: 'verysecurepassword' }
        });

        wrapperLogIn.update();
        expect(wrapperLogIn.state().passwordInput).toEqual('verysecurepassword');

    });

});

describe("Login", () => {


    it("Check login returns error", () => {

        let wrapper = mount(
            <BrowserRouter>
                <LogIn
                    getEntities={jest.fn()}
                    getCategories={jest.fn()}
                />
            </BrowserRouter>);

        let wrapperLogIn = wrapper.find(LogIn).first();

        wrapperLogIn.setState({
            emailInput: "email@test.com",
            passwordInput: "test"
        });

        mockAxios.onPost().reply(400);

        let button = wrapper.find('button#login-button').first();
        button.simulate('submit');

        setImmediate(() => {
            expect(wrapperLogIn.state().alertMessage).toEqual("Error logging in!");
            expect(wrapperLogIn.state().alertType).toEqual("danger");
        });

    });

    it("Check login is successful", () => {

        let wrapper = mount(
            <BrowserRouter>
                <LogIn
                    getEntities={jest.fn()}
                    getCategories={jest.fn()}
                />
            </BrowserRouter>);

        let wrapperLogIn = wrapper.find(LogIn).first();

        wrapperLogIn.setState({
            emailInput: "email@test.com",
            passwordInput: "test"
        });

        mockAxios.onPost().reply(200, {
            token: "nicetoken123"
        });


        let button = wrapperLogIn.find('button#login-button').first();
        button.simulate('submit');

        setImmediate(() => {
            expect(wrapperLogIn.state().alertMessage).toEqual("Successfully logged in!");
            expect(wrapperLogIn.state().alertType).toEqual("success");
        });

    });


});
