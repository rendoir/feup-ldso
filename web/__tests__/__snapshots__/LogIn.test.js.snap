// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check Render ListEvents renders log in form 1`] = `
<div
  id="login-container"
>
  <div
    className="row"
  >
    <div
      className="login-col-first col-sm-6"
    >
      <img
        className=""
        src="logotipo_completo.png"
      />
    </div>
    <div
      className="col-sm-6"
    >
      <form
        className=""
        onSubmit={[Function]}
      >
        <div
          className="input-group"
        >
          <div
            className="input-group-prepend"
          >
            <span
              className="input-group-text"
              id="basic-addon"
            />
          </div>
          <input
            aria-describedby="basic-addon"
            className="email-input form-control"
            onChange={[Function]}
            placeholder="Email"
            required={true}
            type="email"
            value=""
          />
        </div>
        <div
          className="input-group"
        >
          <div
            className="input-group-prepend"
          >
            <span
              className="input-group-text"
              id="basic-addon"
            />
          </div>
          <input
            className="password-input form-control"
            onChange={[Function]}
            placeholder="Password"
            required={true}
            type="password"
            value=""
          />
        </div>
        <button
          className="primary_button btn btn-primary"
          disabled={false}
          id="login-button"
          type="submit"
        >
          Entrar
        </button>
      </form>
    </div>
  </div>
</div>
`;

exports[`renders elements 1`] = `
<nav
  className="navbar navbar-expand navbar-light"
  id="navbar-custom"
>
  <a
    aria-current={null}
    className="navbar-brand"
    href="/events"
    onClick={[Function]}
  >
    <img
      className=""
      id="logo"
      src="/logotipo_agenda.png"
    />
  </a>
  <div
    aria-expanded={null}
    className="navbar-collapse collapse"
    id="responsive-navbar-nav"
  >
    <div
      className="navbar-nav"
      onKeyDown={[Function]}
    />
    <div
      className="navbar-nav"
      onKeyDown={[Function]}
    />
  </div>
</nav>
`;

exports[`renders elements 2`] = `null`;
